### ISSUE TITLE
[Feature request] Add support for cucumber

### ISSUE THREAD

**Issue by tangentially** [presidenten/wdio-video-reporter#21]
**Describe the bug**
When using cucumber as a framework, video recording fails to work. No videos are produced. Cluttered and confusing raw output captured.

**Log**
https://gist.github.com/tangentially/aaf0c0d20364af9cc3279ba1ae120e9d

**To Reproduce**
Create a simple cucumber scenario and run.

**Expected behavior**
Videos recorded, one per scenario.

**Environment (please complete the following information):**
 - **wdio-video-reporter version:** 1.4.4
 - **WebdriverIO version:** 5.12.0
 - **Mode:** WDIO Testrunner
 - **If WDIO Testrunner, running sync/async:** sync
 - **Node.js version:** 10.16.0
 - **NPM version:** 6.10.3
 - **Browser name and version:** Chrome Version 74.0.3729.169 (Official Build) (64-bit)
 - **Platform name and version:** MacOS

**Desktop (please complete the following information):**
 - OS: MacOS
 - Browser Chrome 
 - Version 74

**Additional context**
From wdio-cucumberjs-json-reporter

The order of running of the on* is the following:
onRunnerStart
onSuiteStart (feature)
onSuiteStart (scenario)
onHookStart
onHookEnd
onTestStart
onBeforeCommand
onAfterCommand
onTestPass
onHookStart
onHookEnd
onSuiteEnd (scenario is done)
onSuiteEnd (feature is done)
onRunnerEnd

Please note: onTestEnd() is never called, and that onTestStart() is called once per step. (This leads to a _lot_ of single screenshot directories.

ls rawSeleniumVideoGrabs/
CES-Demo--HAOD-1325-Demo-scenario---subscribe-to-each-package-start-guidance-and-add-waypoints--Given-I-am-on-the-Login-page--CHROME--08-14-2019--11-40-58-918
CES-Demo--HAOD-1325-Demo-scenario---subscribe-to-each-package-start-guidance-and-add-waypoints--Given-I-am-on-the-Login-page--Then-the-password-text-input-is-not-empty--When-I-click-the-Login-button-on-the-Login-page--CHROME--08-14-2019--11-41-04-280
CES-Demo--HAOD-1325-Demo-scenario---subscribe-to-each-package-start-guidance-and-add-waypoints--Given-I-am-on-the-Login-page--When-I-type-drivertest3-username--CHROME--08-14-2019--11-41-00-152
CES-Demo--HAOD-1325-Demo-scenario---subscribe-to-each-package-start-guidance-and-add-waypoints--Given-I-am-on-the-Login-page--When-I-type-drivertest3-username--Then-the-username-text-input-has-value-drivertest3--CHROME--08-14-2019--11-41-01-732

**Comment by vikmaksimenko** [comment]
Same for me, pending for this issue to be resolved

**Comment by presidenten** [comment]
I dont know cucumber. Would you guys mind forking this repo: https://github.com/presidenten/wdio-template
and setting it up with cucumber in a scenario where the bug can be reproduced?

**Comment by vikmaksimenko** [comment]
You may try wdio boilerplate Cucumber project - https://github.com/WarleyGabriel/demo-webdriverio-cucumber. If it doesn't fit for you - I'll be able to fork and update https://github.com/presidenten/wdio-template tomorrow morning (from 9 AM EEST)

**Comment by presidenten** [comment]
A fork would be nice with as slim as possible setup with cucumber so its easy to locate the issue

**Comment by vikmaksimenko** [comment]
@presidenten , I setup wdio-template fork with Cucumber, hope, it would be helpful - https://github.com/vikmaksimenko/wdio-template

**Comment by presidenten** [comment]
Thanks! Ill check it out!

**Comment by presidenten** [comment]
All right... it seems like cucumber constructs many tests from one feature file.

Here is example debug logLevel output:
```

--- New suite: User interactions ---
Incomming command: /session/:sessionId/window/current/position => [position]
Incomming command: /session/:sessionId/window/current/size => [size]

--- New suite: I should be able to edit input (should pass) ---

--- New test: Given I navigate to base url ---
Incomming command: /session/:sessionId/url => [url]
- Screenshot!!

--- New test: Given I open Advanced Examples tab ---
Incomming command: /session/:sessionId/element => [element]
Incomming command: /session/:sessionId/screenshot => [screenshot]
Incomming command: /session/:sessionId/element/0.0004137642967894095-1/click => [click]
- Screenshot!!

--- New test: And I open 'drag-drop-range' demo ---
Incomming command: /session/:sessionId/element => [element]
Incomming command: /session/:sessionId/screenshot => [screenshot]
Incomming command: /session/:sessionId/element/0.0004137642967894095-2/click => [click]
- Screenshot!!
```

and "normal" run where everything stays in one test:
```
--- New suite: User interactions ---

--- New test: should be able to edit input (should pass) ---
Incomming command: /session/:sessionId/url => [url]
- Screenshot!!
Incomming command: /session/:sessionId/element => [element]
Incomming command: /session/:sessionId/screenshot => [screenshot]
Incomming command: /session/:sessionId/element/0.6104175975290829-1/click => [click]
- Screenshot!!
Incomming command: /session/:sessionId/element => [element]
Incomming command: /session/:sessionId/screenshot => [screenshot]
Incomming command: /session/:sessionId/element/0.6104175975290829-2/click => [click]
- Screenshot!!
```

It seems like the first line in the scenario gets its own suite, and all other lines become very small tests in that suite.

Right now Im tracking commands in one single test to make frames. 

To support cucumber would require a different code path where all tests in one suite needs to be concatenated and handled to add frames in the same directory, but the full test-name wont be known until all tests in the suite that corresponds to the scenario has been run, so the directory naming needs some overview and be extended with some meta data to find it again when its time to generate videos. And also framenumbers needs to increase when changing suites, but only after the last test in a suite has passed, which I guess will be a bit tricky to figure out.

It will add a bit of complexity but might be doable with a reporter option that separates the behaviour from the normal reporter code.

Im not sure it will work together with allure though since the reporter there also splits up in different sections:
![image](https://user-images.githubusercontent.com/3883897/63861654-918c6a80-c9ab-11e9-8a1b-3d79ea6ad658.png)
But a video per scenario would still be useful I guess...

To be honest, my motivation to add cucumber support is kind of low at the moment since I have alot to do at work right now, and no real need for it other than that it would be really nice to have cucumber support in the library.
I will most probably not add support in quite some time, if I dont get bored and want to chew on a interesting problem.

I would however happily accept a pull request with the implemented functionality and can help tidy it up if anyone else is interested and have an urgent need.

**Comment by damonpam** [comment]
It would be a nice feature!

**Comment by jonn-set** [comment]
I have pushed changes to make it compatible with Cucumber >v5, please can someone review and give feedback.

**Comment by jonn-set** [comment]
@presidenten hi, could you be able to review my changes, please? It make this cucumber compatible

**Comment by presidenten** [comment]
> @presidenten hi, could you be able to review my changes, please? It make this cucumber compatible

@jonn-set, sorry for the hold up. I really appreciate the work, as many have asked for this.
 
This semester has been really hectic at work and Im trying my best to keep my head above the water and there is just no spare energy right now since I really need to push through these last two days.

If you could wait a few more days, I will try to have a look on it at the beginning of next week when my vacation starts.

**Comment by jonn-set** [comment]
> > @presidenten hi, could you be able to review my changes, please? It make this cucumber compatible
> 
> @jonn-set, sorry for the hold up. I really appreciate the work, as many have asked for this.
> 
> This semester has been really hectic at work and Im trying my best to keep my head above the water and there is just no spare energy right now since I really need to push through these last two days.
> 
> If you could wait a few more days, I will try to have a look on it at the beginning of next week when my vacation starts.

No problem, whenever you have free time, sure can wait till next week. Will give you a nudge next week.

Thanks

**Comment by presidenten** [comment]
v2.0.0 is released and published on npm

Please see [release notes](https://github.com/presidenten/wdio-video-reporter/releases/tag/v2.0.0) and cucumber section in [readme](https://github.com/presidenten/wdio-video-reporter#cucumber-support)
